src/xsdata.h

```diff
@@ -0,0 +1,123 @@
+//! \file xsdata.h
+//! A collection of classes for containing the Multi-Group Cross Section data
+
+#ifndef XSDATA_H
+#define XSDATA_H
+
+#include <cmath>
+#include <cstdlib>
+#include <algorithm>
+#include <vector>
+
+#include "constants.h"
+#include "hdf5_interface.h"
+#include "math_functions.h"
+#include "random_lcg.h"
+#include "scattdata.h"
```

Comment:
Same comment as before. Only include what is needed for interface; move everything else to implementation.

Reply:
- [x]